sdk/Sdk.Generators/FunctionMetadataProviderGenerator.cs

```diff
@@ -0,0 +1,841 @@
+        /// representation of the object, or leaving it as is if the object is a string or Enum type.
+        /// </summary>
+        /// <param name="propValue">The property that needs to be formmated into a string.</param>
+        /// <returns></returns>
+        internal static string FormatObject(object propValue)
+        {
+            if (propValue != null)
+            {
+                // catch values that are already strings or Enum parsing
+                // we don't need to surround these cases with quotation marks
+                if (propValue.ToString().Contains("\"") || propValue.ToString().Contains("Enum"))
+                {
+                    return propValue.ToString();
+                }
+
+                return "\"" + propValue.ToString() + "\"";
+            }
+            else
+            {
+                return "null";
+            }
+        }
+
+        /// <summary>
+        /// Format an array into a string.
+        /// </summary>
+        /// <param name="enumerableValues">An array object to be formatted.</param>
+        /// <returns></returns>
+        internal static string FormatArray(IEnumerable enumerableValues)
+        {
+            string arrAsString;
+
+            arrAsString = "new List<string> { ";
+
+            foreach (var o in enumerableValues)
+            {
+                arrAsString += FormatObject(o);
+                arrAsString += ",";
+            }
+
+            arrAsString = arrAsString.TrimEnd(',', ' ');
+            arrAsString += " }";
+
+            return arrAsString;
+        }
+
+        /// <summary>
+        /// Colllect all of the properties associated with an attribute.
+        /// </summary>
+        /// <param name="attribMethodSymbol">The symbol that represents the attribute constructor method.</param>
+        /// <param name="attributeData">Contains the values associated with the attribute constructor method properties.</param>
+        /// <returns></returns>
+        internal static IDictionary<string, object> GetAttributeProperties(IMethodSymbol attribMethodSymbol, AttributeData attributeData)
+        {
+            Dictionary<string, object> argumentData = new();
+            if (attributeData.ConstructorArguments.Any())
+            {
+                LoadConstructorArguments(attribMethodSymbol, attributeData, argumentData);
+            }
+
+            foreach (var namedArgument in attributeData.NamedArguments)
+            {
+                if (namedArgument.Value.Value != null)
+                {
+                    if (String.Equals(namedArgument.Key, Constants.IsBatchedKey))
+                    {
+                        var argValue = (bool)namedArgument.Value.Value; // isBatched only takes in bools and the generator will parse it as a bool so we can type cast this to use in the next line
+
+                        if (argValue)
+                        {
+                            argumentData["Cardinality"] = "Many";
```

Comment:
I see we have a `Cardinality` enum. Is it worth using that instead of strings?

Reply:
I think the enum you're mentioning is in the main SDK project (which we won't have access to). May be worth to have a copy of that object for this project.